### ISSUE TITLE
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed

### ISSUE THREAD

**Issue by tbenst** [flatironinstitute/CaImAn#820]
- **Tell us a bit about your setup**: 
 1. Operating system (Linux/macOS/Windows): Linux
 
 2. Python version (3.x): 3.7
 
 3. Working environment (Python IDE/Jupyter Notebook/other): Jupyter
 
 4. Which of the demo scripts you're using for your analysis (if applicable): pipeline.ipynb
 
 5. CaImAn version*: 1.8.5
 
 6. CaImAn installation process (`pip install .`/`pip install -e .`/conda): `pip install -e .`
- **Describe the issue that you are experiencing**
Pipeline fails on motion correction step.

options:
```
> opts_dict = {'fnames': gcamp_tifs,
            'fr': fr,
            'decay_time': decay_time,
            'strides': strides,
            'overlaps': overlaps,
            'max_shifts': max_shifts,
            'max_deviation_rigid': max_deviation_rigid,
            'pw_rigid': pw_rigid,
            'p': 1,
            'nb': gnb,
            'rf': rf,
            'K': K, 
            'stride': stride_cnmf,
            'method_init': method_init,
            'rolling_sum': True,
            'only_init': True,
            'ssub': ssub,
            'tsub': tsub,
            'min_SNR': min_SNR,
            'rval_thr': rval_thr,
            'use_cnn': True,
            'min_cnn_thr': cnn_thr,
            'cnn_lowest': cnn_lowest,
            'is3D': False}

> opts = params.CNMFParams(params_dict=opts_dict)
> opts.motion
{'border_nan': 'copy',
 'gSig_filt': None,
 'is3D': False,
 'max_deviation_rigid': 4,
 'max_shifts': (10, 10),
 'min_mov': None,
 'niter_rig': 1,
 'nonneg_movie': True,
 'num_frames_split': 80,
 'num_splits_to_process_els': None,
 'num_splits_to_process_rig': None,
 'overlaps': (24, 24),
 'pw_rigid': True,
 'shifts_opencv': True,
 'splits_els': 6,
 'splits_rig': 6,
 'strides': (48, 48),
 'upsample_factor_grid': 4,
 'use_cuda': False,
 'indices': (slice(None, None, None), slice(None, None, None))}
```

Indices does not look correct...? This does not match expected default from https://github.com/flatironinstitute/CaImAn/issues/712#issuecomment-587161844.

Error:
```
indices---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-40-9630756fb84d> in <module>
     39 
     40     #%% Run piecewise-rigid motion correction using NoRMCorre
---> 41     mc.motion_correct(save_movie=True)
     42     m_els = cm.load(mc.fname_tot_els)
     43     border_to_0 = 0 if mc.border_nan is 'copy' else mc.border_to_0

~/code/CaImAn/caiman/motion_correction.py in motion_correct(self, template, save_movie)
    242 
    243         if self.pw_rigid:
--> 244             self.motion_correct_pwrigid(template=template, save_movie=save_movie)
    245             if self.is3D:
    246                 # TODO - error at this point after saving

~/code/CaImAn/caiman/motion_correction.py in motion_correct_pwrigid(self, save_movie, template, show_template)
    341         if template is None:
    342             logging.info('Generating template by rigid motion correction')
--> 343             self.motion_correct_rigid()
    344             self.total_template_els = self.total_template_rig.copy()
    345         else:

~/code/CaImAn/caiman/motion_correction.py in motion_correct_rigid(self, template, save_movie)
    303                 var_name_hdf5=self.var_name_hdf5,
    304                 is3D=self.is3D,
--> 305                 indices=self.indices)
    306             if template is None:
    307                 self.total_template_rig = _total_template_rig

~/code/CaImAn/caiman/motion_correction.py in motion_correct_batch_rigid(fname, max_shifts, dview, splits, num_splits_to_process, num_iter, template, shifts_opencv, save_movie_rigid, add_to_movie, nonneg_movie, gSig_filt, subidx, use_cuda, border_nan, var_name_hdf5, is3D, indices)
   2718                                                              num_splits=num_splits_to_process, shifts_opencv=shifts_opencv, nonneg_movie=nonneg_movie, gSig_filt=gSig_filt,
   2719                                                              use_cuda=use_cuda, border_nan=border_nan, var_name_hdf5=var_name_hdf5, is3D=is3D,
-> 2720                                                              indices=indices)
   2721         if is3D:
   2722             new_templ = np.nanmedian(np.stack([r[-1] for r in res_rig]), 0)

~/code/CaImAn/caiman/motion_correction.py in motion_correction_piecewise(fname, splits, strides, overlaps, add_to_movie, template, max_shifts, max_deviation_rigid, newoverlaps, newstrides, upsample_factor_grid, order, dview, save_movie, base_name, subidx, num_splits, shifts_opencv, nonneg_movie, gSig_filt, use_cuda, border_nan, var_name_hdf5, is3D, indices)
   2978     dims, T = cm.source_extraction.cnmf.utilities.get_file_size(fname, var_name_hdf5=var_name_hdf5)
   2979     z = np.zeros(dims)
-> 2980     dims = z[].shape
   2981     logging.debug('Number of Splits: {}'.format(splits))
   2982     if type(splits) is int:

IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```

Edit: for what it's worth, I had all tifs as individual files (44,000 of them), so concatenated into one file with:
```
import os
import glob
import caiman as cm
fld = ''  # path to folder where the data is located
fls = glob.glob(os.path.join(fld,'*.tif'))  #  change tif to the extension you need
fls.sort()  # make sure your files are sorted alphanumerically
m = cm.load_movie_chain(fls)
m.save(os.path.join(fld,'data.tif'))
```

Curiously, when I do `tif = tifffile.imread('data.tif')` and call `tif.shape`, I get `(512,512)`, yet the file is 25GB as expected. `ctif = cm.load(gcamp_tifs)` is also only `512x512`

Saving works properly if I do:

```
> tifs = np.array([tifffile.imread(t) for t in fls])
> tifffile.imwrite('gcamp_combined.tif', tifs)
> tifffile.imread('gcamp_combined.tif').shape
(24942, 512, 512)
```

but then `ctif = cm.load(gcamp_tifs)` still only reads in `(512, 512)`
but then

**Comment by tbenst** [comment]
Can confirm everything works fine if I save with tifffile, hence either `load_movie_chain` or `m.save` appears to be broken

**Comment by j-friedrich** [comment]
Works fine on a standard caiman installation where the tifffile version is pinned in environment.yml to 0.15.1. @tbenst are you using a different tifffile version?

**Comment by j-friedrich** [comment]
Fixed in the dev branch (PR #827) which allows usage of newer versions of tifffile.

**Comment by tbenst** [comment]
Great thanks!
